### PR TITLE
Chore: Update CODEOWNERS

### PR DESCRIPTION
**Pull request by zserge**
This PR updates CODEOWNERS so that:
* All directory rules end with `/` (many tools expect that from codeowners syntax)
* Every file in `/pkg/` matches at least one rule and is owned by some team
* Every rule in `CODEOWNERS` matches at least one file
* There is no more `*.go` wildcard for the backend platform

Also had to remove old goconvey file that was not matched and was not needed.

To be improved:
* Make sure that order of the teams is correct because order defines the precedence, not the length or syntax of the rules themselves
* Make sure that in `/public/` folder same rules apply as in `pkg/`, i.e. everything is owned and all rules make sense
* Who owns `/scripts/`?

### CODE REVIEW

**Review by dsotirakis — APPROVED** [review]

File: .github/CODEOWNERS
Comment (ying-jeanne) [inline .github/CODEOWNERS:33]: i saw that you overwrite the subfolders inside of services later, maybe we should remove here, since accesscontrol/alerting/auth/authn inside of services are not ours

Comment (ying-jeanne) [reply .github/CODEOWNERS:33]: just saw you commented it out 👯

File: .github/CODEOWNERS
Comment (ying-jeanne) [inline .github/CODEOWNERS:125]: this is us or plugin?

**Review by ying-jeanne — APPROVED** [review]

File: .github/CODEOWNERS
Comment (zserge) [reply .github/CODEOWNERS:33]: No, thanks, good catch. It's an accidental leftover

**Comment by ying-jeanne** [comment]
would it worthy to define .github/CODEOWNERS to be owned by a group of governor to make sure that in the future, if this file changed, a default group of people (cross teams) could review the changes and get informed? I think now nobody own this file

**Comment by zserge** [comment]
> would it worthy to define .github/CODEOWNERS to be owned by a group of governor to make sure that in the future, if this file changed, a default group of people (cross teams) could review the changes and get informed? I think now nobody own this file

I think we're entering the phase where team structure and package structure get closer, and ownership would become more explicit and clear. This PR is only the first step into that direction. More fixes needed on the frontend package structure/ownership rules etc. Also, some tooling to make sure that CODEOWNERS files matches all the files during the CI is needed. To make this process go faster, I'd suggest not to restrict who own CODEOWNERS for now (to not get blocked).

Maybe release engineering has any ideas on this, @dsotirakis? I'm afraid that if we spread ownership and someone from team A changes ownership of some package from team B to team C and team D approves this - how is that different from "everyone (or no one) owns CODEOWNERS file" as it is now?

**Comment by dsotirakis** [comment]
> Also, some tooling to make sure that CODEOWNERS files matches all the files during the CI is needed. 

What if we use something like this?
https://github.com/grafana/deployment_tools/tree/master/scripts/git/generate-codeowners
This way we can identify if we are touching files that are not owned by anyone, and then throw messages in the PR or something.

> To make this process go faster, I'd suggest not to restrict who own CODEOWNERS for now (to not get blocked).

> I'm afraid that if we spread ownership and someone from team A changes ownership of some package from team B to team C and team D approves this - how is that different from "everyone (or no one) owns CODEOWNERS file" as it is now?

Agree. Until we figure out something I'd rather have people manually requesting review from teams affected rather than something automated, at least for now.

**Comment by sakjur** [comment]
[Wikipedia's Be Bold](https://en.wikipedia.org/wiki/Wikipedia:Be_bold) principle translates rather well here. The consequence of an ill-advised update to this file essentially amounts to some confusion and a revert. We're not shipping this, it's not documentation for end-users that will cause frustration if it's incorrect, and it doesn't affect the builds. I think that as long as you and one more person with merge rights think it's fine to merge a change to this file, it probably is.

**Pull request merged**
